runtime.go

```diff
@@ -21,13 +22,13 @@ type guestRuntime struct {
 	initialized bool
 }
 
-func detectGuestRuntime(ctx context.Context, p *Plugin) guestRuntime {
-	runtime, ok := haskellRuntime(ctx, p, p.module)
+func detectGuestRuntime(p *Plugin) guestRuntime {
+	runtime, ok := haskellRuntime(p, p.mainModule)
```

Comment:
I think the issue with Big Go is that we are really only initializing the main module, dependency are still not being initialized 🤔

Reply:
(working on a patch...)

Reply:
@mhmd-azeez I pushed a commit to this branch (feel free to revert or iterate if you think it's appropriate) with a fix. The issue that I can foresee is that, if there are cross-dependencies between modules, then the init order is random! The "proper" way to init dependencies would be to linearize the dependency graph and then initialize it in that order. We can do that in a follow-up, though!

Reply:
thansk!